### PR TITLE
HTML: window.open() can set opener of a nested browsing context

### PR DESCRIPTION
**Pull request by annevk**
Needed for https://github.com/whatwg/html/pull/4284.

### CODE REVIEW

File: html/browsers/windows/embedded-opener-a-form.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<!doctype html>
+<title>opener and embedded documents; using a and form</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<iframe name=matchesastring></iframe>
+<a href=/common/blank.html target=matchesastring>&lt;a></a>
+<form action=/common/blank.html target=matchesastring><input type=submit value="<form>"></form>
```
Comment (zcorpan) [inline html/browsers/windows/embedded-opener-a-form.html]: The form part is not tested?

Comment (annevk) [reply html/browsers/windows/embedded-opener-a-form.html]: `document.querySelector("input").click();` tests it, no?

**Comment by annevk** [comment]
@foolip @domenic can you look at this?

**Review by domenic — APPROVED** [review]
These tests both seem reasonable, although I don't understand the relationship between them.

**Comment by annevk** [comment]
They're both named targeting, but one of them ends up setting opener and the other doesn't. This was one of the changes with browsing context refactoring, that `window.open()` now directly sets opener sometimes.

**Pull request merged**